<a href="https://colab.research.google.com/github/plorantina/Pemerograman-Dasar/blob/main/LATIHAN%208%20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Laporan Pratikum Dan Latihan**
# **Pemrograman Dasar**

**Dosen Pengajar : Muhammad Veven,S.Kom,M.Eng (125336)**

**Disusun Oleh: Tasya Plorantina Limbong  (4232401030)**

**Teknik Elektro**

**Teknologi Rekayasa Pembangkit Energi**

**Politeknik Negeri Batam**

**2025**



**A.	Langkah- langkah Percobaan Pratikum**

1.	Masuk ke Akun Google Drive
*   Masuk ke akun google, kemudian telusuri website google drive
*   Login akun google drive menggunakan akun email kampus atau email sendiri.
*   Buat folder baru:
    1.   Klik kanan New folder, beri nama formatnya: “Pemerograman Dasar”
    2.   Klik foldernya, kemudian tambah folder lagi : “Pertemuan 3”

2.   Masuk ke Akun Google Colab


*  Klik “Peetemuan 3”, klik kanan kemudian pilih “lainnya”
*  Klik google colab, jika belum ada google cplab, instal terlebih dahulu.
*  File baru otomatis akan terbuka di tab baru.
*  Menulis program pertema di google colab, seperti;
      1.  nama = Tasya
      2.  umur = 20
      3.  jurusan = RPE
      4.  dan lain-lain


*  Ketik “print” untuk menampilkan data pratikum, kemudian klik “Run Cell”.
*  Perhatikan outputnya dibawah cell.
*  Untuk tambah cell baru, klik “+Code”
*  Google colab otomatis menyimpan hasil coding atau perubahan dalam google drive.

3.  Repository ke GitHub
*  Buka website GitHub.Buka website GitHub.
*  Klik tombol “New Repository”.
*  Isi nama repository “Pemerograman dasar”, pilih public, klik tombaol hijau Create repository.
*  Kembali ke google colab, kemudian klik menu file.
*  Pilih “save a copy in GitHub. Klik “OK”
*  Otomatis terhubung dengan app GitHub.
*  Filenya otomatis tersimpan, dan dapat diakses siapapun.







**B.	Peralatan dan Kegunaannya**

1.  Python adalah Bahasa pemrograman yang digunakan untuk belajar coding dasar dan blok kode terstruktur yang dapat digunakan kembali untuk melakukan tugas tertentu.
2. Google Drive kegunaanya adalah sebagai tempat menyimpan file. Google drive dapat menyimpan gambar, video, hasil coding yang kita buat, dokumen ataupun file. Google drive dapat mempermudah mengirim file ke orang sekitar.
3. Google Colaboratory adalah platform cloud yang menjalankan kode Python. Google Colab dapat memudahkan menyimpan hasil kerja langsung ke Google Drive. Google Colab dipakai untuk mengolah data, bikin program machine learning, dan latihan coding.
4. GitHub adalah  platform berbasis web  yang digunakan pengembang untuk menyimpan, mengelola, dan berkolaborasi dalam proyek perangkat lunak.

**C.	Hasil Latihan dan Penjelasannya**

Pada Latihan kali ini, akan menerapkan teknik data wraagling tau pembersihan data dengan menggunakan bebera fungsi atau method yang sudah di bahasa pada teori, jadi hasil nya adalah data yg bersih dan siap untuk di pakai

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

Pada bagian ini, dilakukan persiapan sebelum proses data wrangling atau pembersihan data dilakukan. Beberapa library penting di-import terlebih dahulu agar proses pengolahan data bisa dilakukan dengan baik.
1.  "import numpy as np". Library NumPy digunakan untuk melakukan operasi matematika dan perhitungan numerik.
2.  "import pandas as pd". Pandas adalah library utama untuk pengolahan data dalam bentuk tabel (dataframe).
3.  "import matplotlib.pyplot as plt". Matplotlib adalah library untuk menampilkan grafik dari hasil analisis data.
4.  "import seaborn as sns". Seaborn adalah library visualisasi seperti Matplotlib, tetapi tampilannya lebih menarik dan informatif secara default.

**Memuat tabel**

In [2]:
tasya_df = pd.read_csv('/content/mahasiswa (3).csv')
#import file csv

Perintah pd.read_csv() berfungsi untuk mengimpor data dari file CSV ke dalam DataFrame sehingga data siap diolah lebih lanjut dalam proses analisis atau pembersihan.

**Load tabel**

In [3]:
tasya_df.head()
#melihat 5 data teratas

nim   nama           jurusan    umur nilai_akhir  kehadiran_pct  \
0  E0001   Aldi            Teknik     150        59.5           82.0   
1  E0002   Budi            teknik    20.0        84.5           99.0   
2  E0003  Citra            Energi      -5        56.2           78.0   
3  E0004   Dewi            Teknik  twenty        63.0           99.0   
4  E0005    Eka  Manajemen Energi    22.0         999           65.0   

  tanggal_daftar  
0     2024-05-05  
1     2024-02-14  
2     2024-04-29  
3     2024-08-02  
4     2024-03-22

Hasil penampilan lima data teratas menggunakan fungsi .head(), terlihat bahwa dataset sudah berhasil dimuat namun masih memiliki beberapa masalah dalam kualitas data.
Beberapa nilai pada kolom umur tidak sesuai, seperti adanya angka yang terlalu besar, nilai negatif, hingga format teks yang seharusnya berupa angka ada terdapat ketidakkonsistenan pada penulisan jurusan serta nilai akhir yang tidak realistis.

In [4]:
tasya_df.tail()
#melihat 5 data terakhir

nim   nama      jurusan  umur nilai_akhir  kehadiran_pct tanggal_daftar
73  E0057  Fauzi       Energi  23.0        58.2           76.0     2024-02-05
74  E0039   Mita       Energi  23.0        77.2           79.0     2024-01-04
75  E0069  Salsa  Informatika  23.0        63.4           86.0     2024-04-14
76  E0031   Eros  Informatika  23.0        68.0           66.0     2024-02-27
77  E0026   Zara  Informatika  24.0        90.4           82.0     2024-11-24

Fungsi .tail() digunakan untuk memeriksa data bagian akhir, memastikan apakah struktur data konsisten sampai baris terakhir.
Hasil tampilan lima data terakhir menggunakan fungsi .tail(), terlihat bahwa data pada bagian akhir dataset memiliki format yang lebih rapi dan konsisten dibandingkan baris awal. Seluruh kolom, termasuk umur, nilai akhir, dan jurusan, memiliki nilai yang sesuai tipe datanya. Proses pembersihan data perlu difokuskan untuk baris-baris yang memiliki nilai tidak valid agar keseluruhan dataset menjadi lebih akurat dan siap digunakan dalam analisis.

**Assessing Data**

In [5]:
tasya_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             78 non-null     object 
 1   nama            78 non-null     object 
 2   jurusan         76 non-null     object 
 3   umur            76 non-null     object 
 4   nilai_akhir     76 non-null     object 
 5   kehadiran_pct   76 non-null     float64
 6   tanggal_daftar  76 non-null     object 
dtypes: float64(1), object(6)
memory usage: 4.4+ KB


dari hasil infor terdapat jumlah data yg tidak konsisten ada 2 kolom yg jumlah nya 28 sedangkan yg lainya hanya 76

Perintah tasya_df.info() digunakan untuk menampilkan informasi ringkas mengenai struktur dataset, seperti jumlah baris, jumlah kolom, tipe data tiap kolom, serta jumlah nilai yang tidak kosong (non-null). Berdasarkan hasil yang ditampilkan, dataset memiliki total 78 baris dan 7 kolom. Namun, terlihat bahwa tidak semua kolom memiliki jumlah data lengkap. Kolom nim, nama, dan jurusan memiliki 78 nilai non-null, sedangkan kolom umur, nilai_akhir, dan tanggal_daftar masing-masing hanya memiliki 76 data non-null, sehingga terdapat 2 baris yang memiliki nilai kosong pada kolom-kolom tersebut.

In [6]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Terdapat 5 kolom yang NaN atau missing values

*   Mengecek apakah ada nilai yang kosong (NaN / missing values) di setiap kolom dalam DataFrame bernama tasya_df.
*   Fungsi .isna() akan mengubah data menjadi True jika nilai tersebut kosong, dan False jika tidak kosong
*   Fungsi .sum() kemudian menjumlahkan nilai True tadi (karena True = 1 dan False = 0).

Dataset telah dianalisis menggunakan fungsi isna().sum() untuk mendeteksi adanya missing values. Hasilnya menunjukkan bahwa beberapa kolom memiliki nilai kosong. Terdapat lima kolom yang memiliki masing-masing dua data yang hilang.

In [7]:
print("Jumlah duplikasi: ", tasya_df.duplicated().sum())

Jumlah duplikasi:  8


Lalu terdapat juga 8 data duplikasi

tasya_df.duplicated().sum() digunakan untuk menghitung jumlah baris data yang duplikat dalam dataset. Hasil yang ditampilkan menunjukkan bahwa terdapat 8 baris duplikat. Yang berarti ada data lebih dari satu kali dengan nilai yang sama di semua kolom. Duplikasi memengaruhi kualitas analisis karena dapat menyebabkan hasil statistik atau prediksi menjadi tidak akurat.

**Cleaning Data**

Menghilangkan duplicate data dengan  drop_duplicates()

In [8]:
tasya_df.drop_duplicates(inplace=True)

In [9]:
print("Jumlah duplikasi: ", tasya_df.duplicated().sum())

Jumlah duplikasi:  0


Sekarang data duplikat sudah selesai

Proses pembersihan data telah berhasil dilakukan dengan menggunakan fungsi drop_duplicates() untuk menghapus baris yang memiliki nilai ganda. Setelah fungsi dijalankan dan diperiksa kembali menggunakan duplicated().sum(), hasilnya menunjukkan bahwa sudah tidak ada lagi data duplikat dalam dataset.

**Menangani missing value**

In [10]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Pengecekan missing value menunjukkan bahwa masih terdapat nilai kosong pada beberapa kolom dalam dataset. Kolom yang masih memiliki missing value adalah jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar, masing-masing dengan dua data yang belum terisi, sedangkan kolom nim dan nama tidak memiliki nilai kosong.


In [11]:
tasya_df['jurusan'] = tasya_df['jurusan'].fillna("Pembangkit Energi")

#di sini untuk jurusan di isi dengan "Pembangkit Energi"

Fungsi .fillna() bekerja untuk mengganti nilai kosong (NaN) pada kolom tertentu.
Langkah imputasi yang dilakukan mengisi nilai kosong pada kolom jurusan menggunakan nilai default yaitu "Pembangkit Energi". Dengan pengisian kolom tersebut kini tidak lagi memiliki missing value sehingga dataset menjadi lebih lengkap dan siap digunakan dalam proses analisis data berikutnya.

In [12]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

DataFrame bernama tasya_df awalnya memiliki masalah nilai yang hilang (missing values) pada beberapa kolom. Perintah tasya_df.isna().sum() mengungkapkan bahwa kolom nim, nama, dan jurusan sudah lengkap, namun terdapat 2 nilai yang hilang pada setiap kolom umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar.

In [13]:
tasya_df['umur'] = tasya_df['umur'].fillna(23)

Masalah nilai hilang pada kolom umur diatasi melalui teknik imputasi dengan menjalankan kode tasya_df['umur'] = tasya_df['umur'].fillna(23). Kode ini mengganti 2 nilai yang hilang di kolom umur dengan nilai konstan 23. Kolom umur sudah bersih dari data yang hilang, siap untuk proses analisis lebih lanjut.

In [14]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Data tasya_df menunjukkan bahwa kolom nim, nama, jurusan, dan umur kini sudah bersih dari data yang hilang (0 nilai hilang), mengonfirmasi keberhasilan proses imputasi pada kolom umur. Namun, data data cleaning belum selesai karena masih ada 2 nilai yang hilang pada masing-masing kolom nilai_akhir, kehadiran_pct, dan tanggal_daftar.

In [15]:
tasya_df['nilai_akhir'] = pd.to_numeric(tasya_df['nilai_akhir'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

tasya_df['nilai_akhir'].mean() #mecari rata-rata

np.float64(88.15384615384616)

Kolom nilai_akhir untuk imputasi data hilang. Yang pertama adalah memastikan kebersihan tipe data dengan menggunakan pd.to_numeric(..., errors='coerce') untuk mengonversi semua entri menjadi angka float, sekaligus mengubah data yang tidak valid menjadi nilai yang hilang (NaN). Setelah konversi, selanjutnya menghitung nilai rata-rata (mean) dari kolom nilai_akhir yang bersih, yang mana hasilnya ditemukan sebesar 88.153... Nilai rata-rata ini dipersiapkan untuk digunakan pada langkah selanjutnya untuk mengisi (imputasi) 2 nilai yang hilang pada kolom nilai_akhir, sehingga data menjadi lengkap untuk analisis.

In [16]:
tasya_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     object 
 4   nilai_akhir     65 non-null     float64
 5   kehadiran_pct   68 non-null     float64
 6   tanggal_daftar  68 non-null     object 
dtypes: float64(2), object(5)
memory usage: 4.4+ KB


Jika kita lihat di sini tipe dara pada nilai_akhir sudah berubah menjadi float

Data pada tasya_df mengonfirmasi bahwa DataFrame memiliki total 70 baris. Secara positif, konversi tipe data pada kolom nilai_akhir telah berhasil menjadikannya float64 (angka desimal). Namun, proses konversi ini tampaknya meningkatkan jumlah nilai yang hilang pada kolom tersebut menjadi 5 (65 terisi dari 70 total), selain itu kolom kehadiran_pct dan tanggal_daftar juga masih memiliki 2 nilai yang hilang pada masing-masing kolom. Kolom lain (nim, nama, jurusan, umur) sudah lengkap (70 non-null). Oleh karena itu, langkah selanjutnya yang paling mendesak adalah melakukan imputasi atau penanganan terhadap 5 nilai yang hilang di kolom nilai_akhir dan 2 nilai yang hilang di dua kolom lainnya.

In [17]:
tasya_df['nilai_akhir'] = tasya_df['nilai_akhir'].fillna(88)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Perintah tasya_df['nilai_akhir'] = tasya_df['nilai_akhir'].fillna(88), data yang hilang (NaN) di kolom nilai_akhir secara permanen digantikan dengan nilai 88. Nilai 88 adalah pembulatan dari nilai rata-rata yang telah dihitung pada langkah sebelumnya (88.15...). Setelah eksekusi kode ini, kolom nilai_akhir yang sebelumnya memiliki 5 nilai hilang kini menjadi lengkap dan siap untuk dianalisis lebih lanjut.

In [18]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Proses pembersihan data pada DataFrame tasya_df telah berhasil, menjadikan lima dari tujuh kolom (nim, nama, jurusan, umur, nilai_akhir) sepenuhnya bersih dan bebas dari nilai yang hilang. Keberhasilan ini dicapai melalui serangkaian imputasi, termasuk pengisian kolom nilai_akhir dengan nilai 88. Saat ini, hanya 4 nilai yang hilang yang tersisa pada dua kolom terakhir, yaitu kehadiran_pct dan tanggal_daftar, yang menjadi fokus pembersihan data selanjutnya.

**Lakukan untuk kolom Kehadiran**

In [19]:
tasya_df['kehadiran_pct'] = pd.to_numeric(tasya_df['kehadiran_pct'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

tasya_df['kehadiran_pct'].mean() #mecari rata-rata

np.float64(82.13235294117646)

In [20]:
tasya_df['kehadiran_pct'] = tasya_df['kehadiran_pct'].fillna(82)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Perintah pd.to_numeric(..., errors='coerce') memastikan kolom kehadiran_pct bertipe numerik (float). Nilai non-angka apa pun diubah menjadi NaN (nilai hilang).
Metode .mean() kemudian menghitung nilai rata-rata (rata-rata) dari kolom tersebut.
.fillna(82):  mengganti semua nilai NaN (nilai hilang) yang ada di kolom kehadiran_pct dengan nilai rata-rata yang sudah dibulatkan, yaitu 82.

In [21]:
tasya_df.isna().sum()
#cek hasilnya

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    2
dtype: int64

Perintah tasya_df.isna().sum() digunakan untuk menghitung jumlah total nilai yang hilang (NaN) di setiap kolom.
Proses pembersihan data telah berhasil membawa enam dari tujuh kolom pada DataFrame tasya_df menjadi lengkap (0 nilai hilang).
Kolom tanggal_daftar yang menyisakan 2 nilai yang hilang, menjadikannya satu-satunya kolom yang masih memerlukan penanganan untuk menyelesaikan keseluruhan proses data cleaning.

Lakukan untuk kolom tanggal daftar

In [22]:
tasya_df['tanggal_daftar'] = tasya_df['tanggal_daftar'].fillna('2024-03-24')

# karena tanggal lagir di sini bertipe string jadi langsung masukan saja tidak perlu di cari rata-rata nya atau konversi ke numerik

Perintah .fillna() untuk mengganti semua nilai NaN (nilai hilang) yang ada di kolom tanggal_daftar.
Nilai pengganti yang digunakan adalah string '2024-03-24'.


In [23]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Sekarang tidak ada lagi data missing atau NaN

Hasil akhir dari tasya_df.isna().sum() menunjukkan bahwa DataFrame Anda, tasya_df, kini 100% bersih dari nilai yang hilang (missing values). Semua tujuh kolom—nim, nama, jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar—berhasil diimputasi dan kini masing-masing memiliki 0 nilai hilang. DataFrame Anda telah selesai melalui proses data cleaning dan siap sepenuhnya untuk dianalisis lebih lanjut.

**Menangani inaccurate value**

In [24]:
tasya_df.jurusan.value_counts()

jurusan
Energi               21
Informatika          17
Teknik               12
Manajemen Energi     11
Pembangkit Energi     2
teknik                1
 energi               1
ManajemenEnergi       1
EnErGi                1
Teknik                1
ENG                   1
ENERGI                1
Name: count, dtype: int64

Perintah tasya_df.jurusan.value_counts() yang digunakan untuk menganalisis dan menangani nilai yang tidak akurat (inaccurate values) atau inkonsisten dalam kolom jurusan pada DataFrame tasya_df.
Misalnya, jurusan yang seharusnya hanya 'Energi' malah muncul dalam berbagai versi seperti 'energi', 'EnErGi', dan 'ENERGI'. Hal yang sama juga terjadi pada jurusan 'Teknik'. Komputer menganggap setiap penulisan yang berbeda ini sebagai kategori yang terpisah, padahal mereka adalah jurusan yang sama.

In [25]:
tasya_df.umur.value_counts()

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
twenty         1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64

Terdapat 2 kolom yang datanya tidak akurat atau  inaccurate value

Kolom umur menggunakan value_counts() mengungkapkan bahwa kolom tersebut masih mengandung banyak nilai yang tidak valid atau tidak akurat, meskipun sebelumnya sudah diimputasi dan tidak memiliki nilai hilang (NaN).
Meskipun nilai yang benar-benar hilang (NaN) sudah diisi, sekarang kita menemukan angka yang mustahil (seperti 150, 120, dan -5) dan juga teks yang salah (seperti "twenty" atau "21years").

In [26]:
tasya_df['jurusan'] = tasya_df['jurusan'].str.lower().str.strip()
#ubah menjadi huruf kecil

Kolom jurusan sebagai bagian dari proses membersihkan nilai yang tidak konsisten yang ditemukan sebelumnya.
Tujuan perintah ini untuk menyeragamkan format penulisan pada kolom jurusan sehingga varian-varian seperti 'Energi', 'ENeRGi', dan 'ENERGI' dapat dianggap sebagai satu nilai yang sama.

In [27]:
tasya_df.jurusan.value_counts()

jurusan
energi               24
informatika          17
teknik               14
manajemen energi     11
pembangkit energi     2
manajemenenergi       1
eng                   1
Name: count, dtype: int64

Hasil penghitungan nilai unik (value_counts()) pada kolom jurusan mengubah semua menjadi huruf kecil dan menghilangkan spasi berlebih.
Kategori seperti 'energi' dan 'teknik' yang sebelumnya terpisah kini berhasil digabungkan ke dalam satu kategori baku (total energi menjadi 24 dan teknik menjadi 14).Namun, proses ini menyisakan sedikit pekerjaan rumah, yaitu menggabungkan varian 'manajemen energi' dan 'manajemenenergi' yang terpisah karena spasi, serta menentukan kategori untuk singkatan 'eng'.

In [28]:
#untuk memudahkan replacing maka di buatkan mapping jurusan sesuai dengan yg di inginkan, dengan kata-kata yg mirip di masukan ke jurusan yg seusai

mapping_jurusan = {
    "energi": "Pembangkit Energi",
    " energi ": "Pembangkit Energi",
    "ENERGI": "Pembangkit Energi",
    "EnErGi": "Pembangkit Energi",
    "eng": "Pembangkit Energi",
    "teknik": "Pembangkit Energi",
    "Teknik ": "Pembangkit Energi",
    "teknik ": "Pembangkit Energi",
    "manajemenenergi": "Manajemen Energi",
    "manajemen energi": "Manajemen Energi"
}

mapping_jurusan bertujuan untuk menyeragamkan semua varian penulisan yang berbeda atau salah ketik agar jatuh ke dalam kategori jurusan yang benar.
(seperti 'teknik', 'energi', dan 'eng') menjadi kategori baku yang diinginkan (misalnya, "Pembangkit Energi" dan "Manajemen Energi").

In [29]:
tasya_df["jurusan"] = tasya_df["jurusan"].replace(mapping_jurusan)

In [30]:
tasya_df["jurusan"] = tasya_df["jurusan"].str.title()

#kemablikan menjadi huruf besar

mapping_jurusan untuk mendefinisikan penyeragaman. Misalnya, semua varian penulisan 'Energi' dan 'Teknik' diarahkan menjadi "Pembangkit Energi", dan varian 'manajemenenergi' diarahkan menjadi "Manajemen Energi".
Kode tasya_df["jurusan"].replace(mapping_jurusan) menerapkan penggantian massal sesuai mapping tersebut.
Kode (.str.title()) adalah mengembalikan format penulisan menjadi huruf besar di awal setiap kata.

In [31]:
tasya_df.jurusan.value_counts()

jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64

kode tasya_df.jurusan.value_counts() memberikan ringkasan statistik mengenai distribusi data dalam kolom jurusan. Fungsi ini menghitung secara tepat jumlah kemunculan (frekuensi) dari setiap program studi dalam dataset tasya_df. Hasilnya menunjukkan bahwa Pembangkit Energi adalah jurusan dengan jumlah entri terbanyak, yaitu 41. Dua jurusan lainnya adalah Informatika dengan 17 entri dan Manajemen Energi dengan 12 entri. Secara keseluruhan, data ini mencakup total 70 entri, yang mengindikasikan bahwa jurusan Pembangkit Energi lebih dari separuh keseluruhan data dalam kolom tersebut.

In [32]:
print("Hasil setelah normalisasi dan pemetaan:")
print(tasya_df['jurusan'].value_counts())

Hasil setelah normalisasi dan pemetaan:
jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64


Sekarang untuk inaccurate  value pada kolom jurusan sudah selesai dan tidak da yg type atau tidak terkelompok dengan benar

**Selanjutnya lakuakan untuk kolom nilai **

kode tasya_df.jurusan.value_counts() menghitung secara tepat jumlah kemunculan (frekuensi) dari setiap program studi dalam DataFrame tasya_df.
Dengan hasilnya Pembangkit Energi adalah jurusan dengan jumlah entri terbanyak, yaitu 41, Informatika dengan 17 entri dan Manajemen Energi dengan 12 entri, totalnya 70 entri.

In [33]:
tasya_df.umur.value_counts()

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
twenty         1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64

di sini terlihat beberapa data yg tidak konsisten seperti (150, -5, twenty, 120,NaN_string, 21years

Selanjutnya lakukan seperti pada kolom jurusan , bisa di lakukan mapping atau di ganti satu persatu, karena di sini jumlah data yg inkonsisten hanya satum jadi bisa di replace saja

Kode tasya_df.umur.value_counts() dijalankan untuk memeriksa distribusi dan frekuensi nilai pada kolom umur di DataFrame tasya_df.
Kolom umur memiliki masalah kualitas data yang serius karena adanya nilai-nilai yang tidak konsisten (format salah, nilai tidak valid, atau outlier). Karena jumlah nilai yang tidak konsisten ini sedikit (masing-masing hanya 1 entri), metode yang direkomendasikan adalah melakukan replace (mengganti satu per satu) nilai-nilai yang salah tersebut dengan nilai yang benar atau nilai yang hilang (NaN) sebelum melanjutkan analisis.

In [34]:
tasya_df['umur'] = tasya_df['umur'].replace('twenty', 20)

#bisa di lakuakn replace satu persatu seperti ini
# dari sebelumnya twenty menjadi 20

Fungsi .replace() pada Pandas digunakan untuk mencari nilai 'twenty' dalam kolom umur dan mengubahnya menjadi nilai 20 (sebagai angka).

In [35]:
print(tasya_df['umur'].value_counts())

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
20             1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64


Bisa di lakuakn replace satu persatu seperti di atas atau mapping kembali seperti jurusan sebelumnya

Kode print(tasya_df['umur'].value_counts()) dijalankan kembali untuk memverifikasi perubahan yang terjadi.

Perubahan dalam Output:
Nilai 'twenty' yang sebelumnya ada di daftar telah hilang.
Muncul nilai 20 (angka integer) dengan jumlah hitungan (count) 1. Ini adalah entri baru yang dihasilkan dari penggantian 'twenty'.

In [36]:
"""mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': np.nan,
    -5: np.nan,
    120: np.nan,
    150: np.nan
}
"""
#untuk nilai yang NaN_string, -5, 120, 150 bisa langsung di ganti dengan rata-rata umur atau di jadikan missing value, tetapi jika di jadikan missing value maka setelah ini anda akan mengganti missing value tersebut, jadi sebelum itu lebih baik  di cari rata-rata nya terlebih dahulu lalu di ganti pada mappingan ini

"mapping_umur = {\n    'twenty': 20,\n    '21years': 21,\n    'NaN_string': np.nan,\n    -5: np.nan,\n    120: np.nan,\n    150: np.nan\n}\n"

Tujuannya mengatasi nilai yang tidak konsisten, outlier, dan format yang salah.

Awalnya menggunakan fungsi .replace() untuk mengubah nilai teks seperti 'twenty' menjadi angka 20. Kemudian dilanjutkan dengan metode mapping. Seperti mengoreksi nilai format teks ('21years' menjadi 21).

In [37]:
tasya_df['umur'] = pd.to_numeric(tasya_df['umur'], errors='coerce')
tasya_df['umur'].mean()

np.float64(25.220588235294116)

Sekarang setelah di konversi menjadi numerik maka di dapat nilai rata-rata, dan selanjutnya isikan pada mappinganya

Kolom umur diubah secara paksa menjadi tipe data numerik menggunakan pd.to_numeric(..., errors='coerce'), memastikan konsistensi dan mengubah sisa nilai non-angka menjadi NaN standar. Selanjutnya, dilakukan perhitungan nilai rata-rata (.mean()) dari semua entri umur yang valid, menghasilkan nilai sebesar 25.22 tahun. Nilai rata-rata ini akan berfungsi sebagai nilai imputasi yang digunakan untuk mengganti semua nilai hilang (NaN) yang sebelumnya dihasilkan dari mapping

In [38]:
mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': 25,
    -5: 25,
    120: 25,
    150: 25
}

mapping_umur dibuat ulang atau dimodifikasi dengan nilai pengganti (imputasi) yang spesifik

In [39]:
tasya_df['umur'] = tasya_df['umur'].replace(mapping_umur)

Fungsi .replace(mapping_umur) menerapkan kamus yang baru dibuat ke kolom umur dari DataFrame tasya_df.
Kode ini melakukan Imputasi Mean/Median (Imputasi Rata-rata/Median) dengan nilai 25.

Tahap akhir pembersihan data kolom umur, di mana nilai-nilai yang bermasalah diisi dengan nilai pengganti yang representatif (25).

In [40]:
print("Hasil setelah pemetaan:")
print(tasya_df['umur'].value_counts())
print(f"\nTipe data kolom 'umur' sekarang: {tasya_df['umur'].dtype}")

Hasil setelah pemetaan:
umur
23.0    17
25.0    13
20.0    10
24.0     9
22.0     8
19.0     7
21.0     4
Name: count, dtype: int64

Tipe data kolom 'umur' sekarang: float64


Sekarang kolom umur sudah tidak ada lagi data yg inaccruate,tetapi harus di pastikan lagi seperti di bawah ini

Hasil dari value_counts() menunjukkan bahwa kolom umur kini memiliki nilai-nilai yang konsisten dan valid.

Nilai float64 memastikan bahwa sudah sepenuhnya numerik dan siap untuk perhitungan matematika.

Kolom umur telah melalui proses pembersihan yang ketat dan kini sepenuhnya bersih, valid, dan konsisten. Data ini siap untuk digunakan dalam analisis dan pemodelan statistik lebih lanjut.

In [41]:
tasya_df.isna().sum()
#ternyata terdapat 2 nilai yang NaN pada kolom umur

nim               0
nama              0
jurusan           0
umur              2
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Kode tasya_df.isna().sum() mengungkapkan bahwa, meskipun kolom data lain sudah bersih, kolom umur ternyata masih mengandung dua nilai hilang (NaN) yang terlewat dari proses imputasi sebelumnya. Oleh karena itu, DataFrame tasya_df belum sepenuhnya siap untuk analisis dan memerlukan langkah tambahan untuk mengatasi sisa dua nilai NaN tersebut pada kolom umur.

In [42]:
tasya_df.head(50)
#untuk memastika bisa menggunakan sintaks di atas dengan masukan value baris yang ingin di lihat, cara ini juga bisa di lakukan untuk tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi   NaN         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   

   tanggal_daftar  
0      2024-05-05  
1      2024-02-14  
2      2024-04-29  
3      2024-08-02  
4      2024-03-22  
5      2024-06-21  
6      2024-06-25  
7      2024-10-01  
8      2024-11-17  
9      2024-01-24  
10     05/0

tasya_df.head(50): Fungsi .head() digunakan untuk menampilkan data dari baris pertama hingga baris ke-50 dari DataFrame bernama tasya_df. Ini dilakukan untuk memverifikasi data setelah berbagai proses pembersihan (cleaning) dan imputasi yang telah dilakukan sebelumnya.

In [43]:
tasya_df.tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi   NaN         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054   Cahyo  Pembangkit Energi   NaN         62.6           78.0   
54  E0055    Dina  Pembangkit Energi  24.0         67.5           68.0   
55  E0056     Edo  Pembangkit Energi  23.0         88.0           98.0   
56  E0057   Fauzi  Pembangkit Energi  23.0         58.2           76.0   
57  E0058    Gita  Pembangkit Energi  24.0         89.5           90.0   
58  E0059    Hadi        Informatika  23.0         78.8           70.0   
59  E0060   Ilham        Informatika  21.0         88.0           85.0   
60  E0061    Jaya        Informatika  22.0         60.2           84.0   
61  E0062  Kirana  Pembangkit Energi  25.0         92.4           65.0   
62  E0063   Lutfi  Pembangkit Energi  19.0         76.5           97.0   
63  E0064    Mega  Pembangkit Energi  21.0         90.2           80.0   
64  E0065   Nadia        Informatika  23.0         72.5           99.0   
65  E0066    Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067   Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068    Rafi        Informatika  19.0         84.7           74.0   
68  E0069   Salsa        Informatika  23.0         63.4           86.0   
69  E0070   Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
20     2024-07-04  
21     2024-01-25  
22     2024-11-01  
23     2024-09-27  
24     2024-12-14  
25     2024-11-24  
26    24 Nov 2024  
27     2024-09-13  
28     2024-11-23  
29     2024-10-03  
30     2024

Perintah tasya_df.tail(50), yang merupakan fungsi dari pustaka Pandas untuk menampilkan 50 baris terakhir dari DataFrame tasya_df. Ini dilakukan untuk verifikasi data, terutama bagian akhir data.

Terdapat Missing value pada baris 21 dan 54

untuk itu bisa di lakukan replace lagi dengan memasukan rata-rata umur seperti sebelumnya

In [44]:
tasya_df['umur'].mean() #mencari rata-rata

np.float64(22.426470588235293)

tasya_dff['umur'] memilih kolom bernama 'umur' dari DataFrame bernama tasya_dff.

.mean() adalah metode Pandas yang digunakan untuk menghitung nilai rata-rata (aritmetika) dari semua nilai yang ada dalam kolom yang dipilih.

22.426470588235293 adalah nilai rata-rata yang dihitung dari kolom 'umur'.

np.float64, menunjukkan tipe data dari nilai rata-rata tersebut, yaitu bilangan pecahan presisi ganda (float64).

In [45]:
tasya_df['umur'] = tasya_df['umur'].fillna(22) #isi dengan nilai rata-rata

.fillna(): Ini adalah metode Pandas yang digunakan untuk mengisi nilai-nilai yang hilang (NaN/Not a Number) dalam suatu(kolom).

22: Nilai yang dimasukkan ke dalam metode fillna() adalah 22. Angka ini merupakan pembulatan ke bawah dari nilai rata-rata yang dihitung sebelumnya ($\approx 22.42$).

In [46]:
tasya_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Setelah di cek sudah tidak ada lagi data yang missing pada semua kolom data

.isna(): Ini adalah metode Pandas yang diterapkan pada DataFrame. Metode ini akan memeriksa setiap sel dalam DataFrame.
.sum(): Metode ini diterapkan pada hasil dari .isna(). Dalam konteks Boolean (True dan False), True dihitung sebagai 1 dan False dihitung sebagai 0. Oleh karena itu, .sum() akan menjumlahkan semua nilai True di setiap kolom. Jumlah ini secara efektif menunjukkan total nilai yang hilang di masing-masing kolom.

In [47]:
tasya_df.duplicated().sum()

np.int64(0)

Untuk pulikat juga sama tidak ada data yang duplikat

tasya_df.duplicated(): Mengidentifikasi setiap baris yang merupakan duplikat (menghasilkan True untuk duplikat, False untuk yang unik).
.sum(): Menghitung total nilai True (di mana True dihitung sebagai 1), yang berarti menghitung total baris duplikat.

Untuk memverifikasi bisa show head() dan tail() seperti di bawah ini atau count duplicates() dan isnan()

In [48]:
tasya_df.head(55)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi  22.0         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054 

Tujuan Kode tasya_df.head(55) bertujuan untuk menampilkan 55 baris pertama dari DataFrame tasya_df.

In [49]:
tasya_df.tail()

nim   nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
65  E0066   Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067  Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068   Rafi        Informatika  19.0         84.7           74.0   
68  E0069  Salsa        Informatika  23.0         63.4           86.0   
69  E0070  Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
65     2024-01-05  
66     2024-05-15  
67     2024-05-19  
68     2024-04-14  
69     2024-02-17

Kode tasya_df.tail() adalah fungsi bawaan dari Pandas, sebuah library populer di Python untuk analisis data. Fungsi ini digunakan untuk menampilkan beberapa baris terakhir dari sebuah objek DataFrame (yang dalam kasus ini bernama tasya_df).

In [50]:
tasya_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     float64
 4   nilai_akhir     70 non-null     float64
 5   kehadiran_pct   70 non-null     float64
 6   tanggal_daftar  70 non-null     object 
dtypes: float64(3), object(4)
memory usage: 4.4+ KB


JIka di lihat pada info, maka semua jumlah data sama yaitu 70, maka proses Pembersihan data sudah selesai

Kode tasya_df.info() adalah fungsi penting dalam library Pandas di Python yang digunakan untuk mendapatkan ringkasan ringkas dari sebuah DataFrame (tasya_df).
Kolom Non-Null Count untuk semua 7 kolom menunjukkan angka 70, yang sama dengan jumlah total baris (70 entries), ini mengonfirmasi bahwa tidak ada missing value (nilai NaN atau null) di seluruh dataset. Ini berarti tahap penanganan nilai yang hilang (missing data handling) dari Pembersihan Data (Data Cleaning) telah berhasil diselesaikan atau memang tidak diperlukan.

Export data yang sudah di bersihkan menjadi file csv, menggunakan script berikut ini

In [51]:
tasya_df.to_csv("mahasiswa_clean.csv", index=False)

tasya_df.to_csv("mahasiswa_clean.csv", index=False), berfungsi untuk mengubah atau mengekspor data yang ada di dalam objek DataFrame menjadi sebuah file CSV (Comma Separated Values) baru.

Data mahasiswa yang sudah dipastikan lengkap (tidak ada missing value seperti yang ditunjukkan oleh tasya_df.info()) dan siap untuk dianalisis, kini disimpan secara permanen ke dalam file eksternal bernama mahasiswa_clean.csv.

Lalu lakukan Dodownload file csv yang sudah di export sebelumnya

In [52]:
from google.colab import files
files.download("mahasiswa_clean.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Bandingkan hasil data csv sebelumnya dengan csv yang sudah di bersihkan
bisa di buka menggunakan excel / spreadsheet untuk membandingaknya

from google.colab import files: Baris ini mengimpor modul (kumpulan fungsi) bernama files yang disediakan khusus oleh lingkungan Google Colab. Modul ini wajib digunakan untuk berinteraksi dengan sistem file lokal pengguna, termasuk fungsi untuk mengunggah (upload) dan mengunduh (download) file.

**Silahkan di jelskan langkah-langkah dan hasil latihan pada laporan**

---

In [53]:
test_df = pd.read_csv("/content/sample_data/california_housing_train.csv")

In [54]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17000 entries, 0 to 16999
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           17000 non-null  float64
 1   latitude            17000 non-null  float64
 2   housing_median_age  17000 non-null  float64
 3   total_rooms         17000 non-null  float64
 4   total_bedrooms      17000 non-null  float64
 5   population          17000 non-null  float64
 6   households          17000 non-null  float64
 7   median_income       17000 non-null  float64
 8   median_house_value  17000 non-null  float64
dtypes: float64(9)
memory usage: 1.2 MB


Kolom Non-Null Count untuk semua 9 kolom menunjukkan angka 17000, yang sama dengan jumlah total entri. Ini berarti tidak ada missing value (nilai yang hilang atau NaN) dalam dataset ini.

Semua kolom bertipe float64 (angka desimal), yang sesuai untuk data numerik yang akan digunakan dalam pemodelan prediksi.

In [55]:
test_df.describe()

longitude      latitude  housing_median_age   total_rooms  \
count  17000.000000  17000.000000        17000.000000  17000.000000   
mean    -119.562108     35.625225           28.589353   2643.664412   
std        2.005166      2.137340           12.586937   2179.947071   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.790000     33.930000           18.000000   1462.000000   
50%     -118.490000     34.250000           29.000000   2127.000000   
75%     -118.000000     37.720000           37.000000   3151.250000   
max     -114.310000     41.950000           52.000000  37937.000000   

       total_bedrooms    population    households  median_income  \
count    17000.000000  17000.000000  17000.000000   17000.000000   
mean       539.410824   1429.573941    501.221941       3.883578   
std        421.499452   1147.852959    384.520841       1.908157   
min          1.000000      3.000000      1.000000       0.499900   
25%        297.000000    790.000000    282.000000       2.566375   
50%        434.000000   1167.000000    409.000000       3.544600   
75%        648.250000   1721.000000    605.250000       4.767000   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        17000.000000  
mean        207300.912353  
std         115983.764387  
min          14999.000000  
25%         119400.000000  
50%         180400.000000  
75%         265000.000000  
max         500001.000000

Fungsi test_df.describe() pada Pandas digunakan untuk menghasilkan statistik deskriptif dari semua kolom numerik di dalam DataFrame test_df.

In [56]:
test_df.head(60)

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0     -114.31     34.19                15.0       5612.0          1283.0   
1     -114.47     34.40                19.0       7650.0          1901.0   
2     -114.56     33.69                17.0        720.0           174.0   
3     -114.57     33.64                14.0       1501.0           337.0   
4     -114.57     33.57                20.0       1454.0           326.0   
5     -114.58     33.63                29.0       1387.0           236.0   
6     -114.58     33.61                25.0       2907.0           680.0   
7     -114.59     34.83                41.0        812.0           168.0   
8     -114.59     33.61                34.0       4789.0          1175.0   
9     -114.60     34.83                46.0       1497.0           309.0   
10    -114.60     33.62                16.0       3741.0           801.0   
11    -114.60     33.60                21.0       1988.0           483.0   
12    -114.61     34.84                48.0       1291.0           248.0   
13    -114.61     34.83                31.0       2478.0           464.0   
14    -114.63     32.76                15.0       1448.0           378.0   
15    -114.65     34.89                17.0       2556.0           587.0   
16    -114.65     33.60                28.0       1678.0           322.0   
17    -114.65     32.79                21.0         44.0            33.0   
18    -114.66     32.74                17.0       1388.0           386.0   
19    -114.67     33.92                17.0         97.0            24.0   
20    -114.68     33.49                20.0       1491.0           360.0   
21    -114.73     33.43                24.0        796.0           243.0   
22    -114.94     34.55                20.0        350.0            95.0   
23    -114.98     33.82                15.0        644.0           129.0   
24    -115.22     33.54                18.0       1706.0           397.0   
25    -115.32     32.82                34.0        591.0           139.0   
26    -115.37     32.82                30.0       1602.0           322.0   
27    -115.37     32.82                14.0       1276.0           270.0   
28    -115.37     32.81                32.0        741.0           191.0   
29    -115.37     32.81                23.0       1458.0           294.0   
30    -115.38     32.82                38.0       1892.0           394.0   
31    -115.38     32.81                35.0       1263.0           262.0   
32    -115.39     32.76                16.0       1136.0           196.0   
33    -115.40     32.86                19.0       1087.0           171.0   
34    -115.40     32.70                19.0        583.0           113.0   
35    -115.41     32.99                29.0       1141.0           220.0   
36    -115.46     33.19                33.0       1234.0           373.0   
37    -115.48     32.80                21.0       1260.0           246.0   
38    -115.48     32.68                15.0       3414.0           666.0   
39    -115.49     32.87                19.0        541.0           104.0   
40    -115.49     32.69                17.0       1960.0           389.0   
41    -115.49     32.67                29.0       1523.0           440.0   
42    -115.49     32.67                25.0       2322.0           573.0   
43    -115.50     32.75                13.0        330.0            72.0   
44    -115.50     32.68                18.0       3631.0           913.0   
45    -115.50     32.67                35.0       2159.0           492.0   
46    -115.51     33.24                32.0       1995.0           523.0   
47    -115.51     33.12                21.0       1024.0           218.0   
48    -115.51     32.99                20.0       1402.0           287.0   
49    -115.51     32.68                11.0       2872.0           610.0   
50    -115.52     34.22                30.0        540.0           136.0   
51    -115.52     33.13                18.0       11

Kode test_df.head(60) adalah perintah dari library Pandas yang digunakan untuk menampilkan beberapa baris pertama dari sebuah objek DataFrame yang bernama test_df.


Pratikum 8

Link Google Colab : https://colab.research.google.com/drive/1pzZ0q_CWMupyRmmaTZQYjY0my-U7411h#scrollTo=skX0WwptA-M_

Link GitHub:https://github.com/plorantina/Pemerograman-Dasar/blob/main/PRATIKUM%208%20.ipynb



Latihan 8

Link Goohle Colab: https://colab.research.google.com/drive/1ieIDPcozs5W-chIwAA7KCf49NAJ-eL-f#scrollTo=piJkx8pWWgr5

Link GitHub: https://github.com/plorantina/Pemerograman-Dasar/blob/main/Latihan%208%20.ipynb